src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -369,8 +427,13 @@ void EIRPlantLoopHeatPump::doPhysicsASHP(EnergyPlusData &state, Real64 currentLo
     //  calculate power usage from EIR curves
     this->calcPowerUsage(state);
 
-    // evaluate the source side heat transfer rate
-    this->calcSourceSideHeatTransferASHP(state);
+    if (this->heatRecoveryIsActive) {
+        // evaluate the heat recovery side heat transfer rate
+        this->calcHeatRecoveryHeatTransferASHP(state);
+    } else {
+        // evaluate the source side heat transfer rate
+        this->calcSourceSideHeatTransferASHP(state);
+    }
```

Comment:
If the heat recovery is active `calcHeatRecoveryHeatTransferASHP()` function does the source side and heat recovery side calculations; otherwise, the only the source side calculation is performed using `calcSourceSideHeatTransferASHP()` function.